templates/taxonomy-topic.php

```diff
@@ -23,35 +66,44 @@
 			<div class="<?php UCF_FAQ_Config::add_athena_attr( 'row' ); ?>">
 				<div class="<?php UCF_FAQ_Config::add_athena_attr( 'col-md-8' ); ?>">
 			<?php
-			while ( have_posts() ) {
-				the_post();
-				$question_classes = " mt-3 h4";
-				$question_attrs = ' data-toggle="collapse" href="#post' . $post->ID . '"';
-				$answer_classes = " mt-2 mb-4 collapse" . $show;
-				$answer_attrs = ' id="post' . $post->ID . '"';
-			?>
-				<a href="<?php echo get_permalink( $post->ID ); ?>">
-					<h2 class="ucf-faq-question<?php UCF_FAQ_Config::add_athena_attr( $question_classes ); ?>"<?php UCF_FAQ_Config::add_athena_attr( $question_attrs ); ?>>
-						<?php the_title(); ?>
-					</h2>
-				</a>
-				<div class="ucf-faq-topic-answer<?php UCF_FAQ_Config::add_athena_attr( $answer_classes ); ?>"<?php UCF_FAQ_Config::add_athena_attr( $answer_attrs ); ?>>
-					<?php the_content(); ?>
-				</div>
-			<?php
-			}
+				while ( have_posts() ) {
+					the_post();
+
+					// Save FAQ and tag list to filter out of related FAQs section
+					array_push( $faqs, $post );
+
+					foreach( wp_get_post_tags( $post->ID ) as $tag ) {
+						if( !in_array( $tag, $tags ) ) {
+							array_push( $tags, $tag->slug );
+						}
+					}
+
+					echo display_faq( $post, $show, 'h4' );
+				}
+
+				$related_posts = get_related_posts( $tags );
+				$related_faq_html = null;
+
+				foreach( $related_posts as $post ) {
+					if( !in_array( $post, $faqs ) ) { // Don't display FAQs already on the page
```

Comment:
Instead of having to do this if-statement against existing posts in the `$faqs` array here, you could alternatively update your `get_related_posts()` function to include the `post__not_in` argument, which could filter out all the posts in `$faqs` for you.

You'll need to update line 73 to push `$post->ID` instead of `$post` to `$faqs`, and add a second argument to `get_related_posts()` to be able to pass in `$faqs` to do this.

See https://codex.wordpress.org/Class_Reference/WP_Query#Post_.26_Page_Parameters